subprojects/core/src/main/java/org/gradle/internal/model/CalculatedValueContainer.java

```diff
@@ -254,4 +256,22 @@ private void releaseLock() {
             lock.unlock();
         }
     }
+
+    private static class FallbackContext implements NodeExecutionContext {
```

Comment:
💭 It's kind of weird that you have to call `isFallbackContext` on a type named `FallbackContext` to determine that it is a fallback context, which only means "don't emit a build operation for this transform".

Are there better names available here?  Maybe `DelegatingExecutionContext` and `emitsVisibleOperations()` or something?

Reply:
I changed the names and added some Javadoc. PTAL!

Reply:
I think this is much clearer, thanks!